server/middleware/featureFlags.test.ts

```diff
@@ -1,54 +1,195 @@
+      query: { testFlag: queryEnabled ? '1' : '0' },
     })
+
+  let resultRes: Response
+  const next = jest.fn()
+
+  beforeEach(() => {
+    resultRes = mockRes()
   })
 
-  it('overrides a default of true if flag is "0" in request cookies', () => {
-    const req = mockReq({ cookies: { testFlag: '0' } })
-    const res = mockRes()
-    const next = jest.fn()
-    readFeatureFlags({ testFlag: { default: true, description: 'TestDesc', label: 'Test' } })(req, res, next)
-    expect(res.locals.flags).toEqual({
-      testFlag: false,
+  describe('Uses the default if flag not present in request query or cookies', () => {
+    ;[true, false].forEach(def => {
+      it(`- Default value: ${def}`, () => {
+        const req = neitherReq()
+        readFeatureFlags(testFlag(def))(req, resultRes, next)
+        expect(resultRes.locals.flags).toEqual({
+          testFlag: def,
+        })
+      })
     })
   })
 
-  it('uses request query over cookies', () => {
-    const req = mockReq({ query: { testFlag: '0' }, cookies: { testFlag: '1' } })
-    const res = mockRes()
-    const next = jest.fn()
-    readFeatureFlags({ testFlag: { default: true, description: 'TestDesc', label: 'Test' } })(req, res, next)
-    expect(res.locals.flags).toEqual({
-      testFlag: false,
+  describe('When environment feature settings override is not enabled', () => {
+    invalidFeatureDateValueCases.forEach(({ name, value }) => {
+      let featureDisabledProcessEnv: NodeJS.ProcessEnv
+      beforeAll(() => {
+        process.env[featureKey(testFlagKey)] = `${value}`
+        process.env[flagQueriesEnabedKey] = `${true}`
+        featureDisabledProcessEnv = process.env
+      })
+      afterAll(() => {
+        process.env = INITAL_ENV
+      })
+      describe(`- Not enabled value: ${name} - ${value}`, () => {
+        it('Overrides a default of true if flag is "0" in request query', () => {
+          readFeatureFlags(testFlag(true))(queryReq(false), resultRes, next)
+          expect(resultRes.locals.flags).toEqual({
+            testFlag: false,
+          })
+        })
+
+        it('Overrides a default of true if flag is "0" in request cookies', () => {
+          readFeatureFlags(testFlag(true))(cookieReq(false), resultRes, next)
+          expect(resultRes.locals.flags).toEqual({
+            testFlag: false,
+          })
+        })
+
+        it('Uses request query over cookies', () => {
+          readFeatureFlags(testFlag(true))(cookieAndQueryReq(true, false), resultRes, next)
+          expect(resultRes.locals.flags).toEqual({
+            testFlag: false,
+          })
+        })
+
+        it('Overrides a default of false if flag is "1" in request query', () => {
+          readFeatureFlags(testFlag(false))(queryReq(true), resultRes, next)
+          expect(resultRes.locals.flags).toEqual({
+            testFlag: true,
+          })
+        })
+
+        describe('Returns the cokkie/query value regardless of other flag values', () => {
```

Comment:
I think I've lost track a bit here, not sure if I'm misunderstanding or if something you intended is missing: what do you mean here by "regardless of other flag values"? Set in `process.env`? Or somewhere else?

Also, there's a typo in cokkie.